### PR TITLE
Use `windows-2019` image in CI

### PR DESCRIPTION
**Pull request by ddalcino**
As you can see in a recent Azure Pipelines build, the Windows Server 2016 image is now deprecated: https://dev.azure.com/miurahr/github/_build/results?buildId=4687&view=logs&j=c4497b2b-649a-591e-539b-0d716883e33e

The newer `vmImage: 'windows-2022'` would work too; see https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml

### CODE REVIEW

**Comment by ddalcino** [comment]
https://dev.azure.com/miurahr/github/_build/results?buildId=4688&view=logs&j=a846d25a-e32c-5640-1b53-e815fab94407&t=e95a39fa-5cd2-5eb3-7a94-1fe6d1d1b7b0&l=17:
```
FullyQualifiedErrorId : No instances of Visual Studio 2017 found.
```

Looks like all the Windows BuildJob architectures will need to be updated to 2019. Not as simple as I had hoped.

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     - name: Install tools
       run: |
         pip install -U pip
-        pip install tox tox-gh-actions coveralls
+        pip install tox tox-gh-actions coveralls coverage[toml]
```
Comment (ddalcino) [inline .github/workflows/check.yml:21]: This is derived from #446; I can't get the `check.yml` action to pass without it.

File: ci/generate_azure_pipelines_matrices.py [hunk 2]
```diff
@@ -4,12 +4,13 @@
...
 import json
 import random
+import re
 from itertools import product
 from typing import Dict, Optional
 
 MIRRORS = [
     "https://ftp.jaist.ac.jp/pub/qtproject",
-    "http://ftp1.nluug.nl/languages/qt",
+    "https://ftp1.nluug.nl/languages/qt",
```
Comment (ddalcino) [inline ci/generate_azure_pipelines_matrices.py:13]: This is not directly related to the PR, but it should be done nonetheless.

File: ci/generate_azure_pipelines_matrices.py [hunk 3]
```diff
@@ -95,20 +104,21 @@ def __init__(self, platform, build_jobs):
...
         BuildJob(
             "install-qt",
-            "5.14.2",
+            "5.15.2",
             "windows",
             "desktop",
-            "win64_msvc2017_64",
-            "msvc2017_64",
+            "win32_msvc2019",
+            "msvc2019",
```
Comment (ddalcino) [inline ci/generate_azure_pipelines_matrices.py:111]: Qt 5.14.2 has 6 architectures available: `win32_mingw73`, ~~`win32_msvc2017`~~, `win64_mingw73`, `win64_msvc2015_64`, ~~`win64_msvc2017_64`~~, and `wasm_32`. On the Windows-2019 host, MSVC 2017 is no longer available. I have updated the other MSVC 2017 builds to use the other mingw architectures. At this point, all of the architectures available for Qt 5.14.2 are being tested. I didn't know what else to do for this build, so I added the only architecture that wasn't being tested for MSVC 2019, and I changed this build to Qt 5.16.2.

If you disagree with any of the build jobs being tested here, please suggest alternatives and I will be happy to test them.

**Comment by miurahr** [comment]
I'd like to squash and merge it.

**Comment by ddalcino** [comment]
> I'd like to squash and merge it.

Ok, it's squashed.
